include/unifex/let_value.hpp

```diff
@@ -404,17 +375,14 @@ class _sender<Predecessor, SuccessorFactory>::type {
         static_cast<Receiver&&>(receiver)};
   }
 
-  friend constexpr auto tag_invoke(tag_t<unifex::blocking>, const type&) noexcept {
-    constexpr blocking_kind succ = successor_types<_let_v::max_blocking_kind>{}();
-    if constexpr (
-        blocking_kind::never == cblocking<Predecessor>() || blocking_kind::never == succ) {
-      return blocking_kind::never;
-    } else if constexpr (
-        blocking_kind::maybe != cblocking<Predecessor>() && blocking_kind::maybe != succ) {
-      return blocking_kind::constant<_let_v::_blocking_kind(cblocking<Predecessor>(), succ)>{};
-    } else {
-      return _let_v::_blocking_kind(cblocking<Predecessor>(), succ);
-    }
+  friend constexpr blocking_kind tag_invoke(tag_t<unifex::blocking>, const type& sender) noexcept {
+    // get the runtime blocking_kind for the predecessor
+    blocking_kind pred = blocking(sender.pred_);
```

Comment:
This must be `blocking_kind pred = unifex::blocking(sender.pred_);`. This also affects several other files. See also https://github.com/facebookexperimental/libunifex/issues/555

Otherwise:

```
[build] C:\include\unifex/let_value.hpp(406): error C2064: term does not evaluate to a function taking 1 arguments
[build] C:\include\unifex/let_value.hpp(406): note: class does not define an 'operator()' or a user defined conversion operator to a pointer-to-function or reference-to-function that takes appropriate number of arguments
[build] C:\include\unifex/let_value.hpp(406): note: while trying to match the argument list '(const Predecessor)'
[build]         with
[build]         [
[build]             Predecessor=unifex::_just::_sender<>::type
[build]         ]
[build] C:\include\unifex/let_value.hpp(406): note: the template instantiation context (the oldest one first) is
[build] C:\main.cpp(5): note: while compiling class template member function 'unifex::_let_v::_sender<unifex::_just::_sender<>::type,a::<lambda_1>>::type unifex::_let_v::_cpo::_fn::operator ()<unifex::_just::_sender<>::type,a::<lambda_1>>(Predecessor &&,SuccessorFactory &&) noexcept const'
[build]         with
[build]         [
[build]             Predecessor=unifex::_just::_sender<>::type,
[build]             SuccessorFactory=a::<lambda_1>
[build]         ]
[build] C:\include\unifex/let_value.hpp(419): note: see reference to variable template 'const bool is_nothrow_constructible_v<unifex::_let_v::_sender<unifex::_just::_sender<>::type,`a'::`2'::<lambda_1> >::type,unifex::_just::_sender<>::type,`a'::`2'::<lambda_1> >' being compiled
[build] C:\include\unifex/let_value.hpp(317): note: while compiling class 'unifex::_let_v::_sender<unifex::_just::_sender<>::type,a::<lambda_1>>::type'
[build] C:\include\unifex/let_value.hpp(404): note: while compiling class template member function 'unifex::_block::blocking_kind unifex::_let_v::tag_invoke(unifex::_block::_fn,const unifex::_let_v::_sender<unifex::_just::_sender<>::type,a::<lambda_1>>::type &) noexcept'
[build] C:\include\unifex/tag_invoke.hpp(34): note: see the first reference to 'unifex::_let_v::tag_invoke' in 'unifex::_tag_invoke::_fn::operator ()'
```